posts/2024-05-13-jakarta-nosql-challenge-switching-nosql-easily.adoc

```diff
@@ -0,0 +1,258 @@
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Jakarta NoSQL in Action: Switching NoSQL Databases with Ease
+Maximillian Arruda <https://github.com/dearrudam>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+
+// // // // // // // //
+// In the preceding section:
+// Do not insert any blank lines between any of the lines.
+//
+// "open-graph-image" is set to OL logo. Whenever possible update this to a more appropriate/specific image (For example if present a image that is being used in the post). However, it
+// can be left empty which will set it to the default
+//
+// "open-graph-image-alt" is a description of what is in the image (not a caption). When changing "open-graph-image" to
+// a custom picture, you must provide a custom string for "open-graph-image-alt".
+//
+// Replace DESCRIPTION with a short summary (~60 words) of the release (a more succinct version of the first paragraph of the post).
+//
+// If adding image into the post add :
+// -------------------------
+// [.img_border_light]
+// image::img/blog/FILE_NAME[IMAGE CAPTION ,width=70%,align="center"]
+// -------------------------
+// "[.img_border_light]" = This adds a faint grey border around the image to make its edges sharper. Use it around screenshots but not
+// around diagrams. Then double check how it looks.
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+Nowadays, in the cloud era, switching between NoSQL solutions could be interesting to save resources and costs, but it could be a nightmare, specially if we're highly coupled to a particular vendor. It will cause some changes that will make this approach very expensive, like: time spent on the change, the learning curve of a new API to use with this database, the code that will be lost, the persistence layer that needs to be replaced, etc.
+
+This is where the flexibility of Jakarta NoSQL with an implementation like https://www.jnosql.org[Eclipse JNoSQL] together shines.
+
+NOTE: As of March 2024, Eclipse JNoSQL supports about https://www.jnosql.org/docs/supported_dbs.html[30 NoSQL databases].
+
+== Jakarta NoSQL in Action: Switching NoSQL Databases with Ease
+
+To explore this capability, let's use the link:../../../2024/05/06/jakarta-nosql-in-action-meet-jnopo-game.html[*JNopo* game].
+
+Here's the challenge: let's suppose that it's needed to change the *JNopo*'s Document NoSQL database, *MongoDB*, to another Document NoSQL database for save money purposes, e.g. *Couchbase*.
```

Comment:
```suggestion
In our scenario, let's suppose that we need to change the *JNopo*'s Document NoSQL database from *MongoDB*, to another Document NoSQL database (e.g. *Couchbase*) to enable us to save money. The good thing is that because we've already used the Jakarta NoSQL specification to connect to our NoSQL solutions in this application, with just a few small changes we'll be able to achieve this goal easily. Let's take a look at the steps we need to take for this.
```